tap_workday_raas/discover.py

```diff
@@ -81,14 +84,17 @@ def generate_schema_for_report(xsd):
         # When elem's type attribute is a type defined as its own complexType - a nested object
         if elem_type in complex_type_mapping:
 
-            # TODO Check if the element is a list and type it correctly
-            is_list = elem.attrib.get("maxOccurs") == 'unbounded'
+            max_occurs = elem.attrib.get("maxOccurs")
+            if max_occurs not in (None, "unbounded"):
+                raise Exception("Found unexpected value for maxOccurs attribute.")
```

Comment:
Again, it should probably put the value in the exception here for traceability.

Reply:
Good call. Included it.